File: packages/neptune_flutter_ui/lib/src/widgets/neptune_receipt.dart
// © 2026 Neptune.Fintech (neptune.ly) · Neptune Odyssey Community License v1.0

import 'package:flutter/material.dart';

import '../theme/accessibility.dart';
import '../theme/extensions.dart';
import '../theme/neptune_theme.dart';
import 'neptune_buttons.dart';

/// A centred success hero (web `<npt-success>`): a large success-coloured
/// circular check, a display-font title, an optional big amount in money style,
/// and an optional subtitle. Announced as one live stop - "success: Transfer
/// sent, 1,250.000 Libyan dinars, to Ahmed" - the moment it appears.
/// Theme-only, RTL-safe.
class NeptuneSuccess extends StatelessWidget {
  final String title;
  final String? amount;
  final String? subtitle;
  final IconData icon;

  /// The ISO code or symbol for the spoken amount when [amount] carries none.
  final String? currency;

  const NeptuneSuccess({
    super.key,
    required this.title,
    this.amount,
    this.subtitle,
    this.icon = Icons.check_circle,
    this.currency,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final npt = Theme.of(context).extension<NptColors>()!;
    final type = Theme.of(context).extension<NptType>()!;
    final text = Theme.of(context).textTheme;
    final money = NeptuneTheme.moneyStyle(context, base: text.displaySmall)
        .copyWith(color: scheme.onSurface, fontWeight: FontWeight.w700);
    final strings = NeptuneAccessibility.of(context);

    final spoken = [
      '${strings.success}: $title',
      if (amount != null)
        NeptuneAccessibility.money(context, amount!, currency: currency),
      if (subtitle != null) subtitle!,
    ].join(', ');

    return Padding(
      padding: const EdgeInsetsDirectional.all(24),
      child: Semantics(
        liveRegion: true,
        label: spoken,
        excludeSemantics: true,
        child: Column(
        mainAxisSize: MainAxisSize.min,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          Container(
            width: 96,
            height: 96,
            decoration: BoxDecoration(
              color: npt.successContainer,
              shape: BoxShape.circle,
            ),
            child: Icon(icon, size: 56, color: npt.success),
          ),
          const SizedBox(height: 16),
          Text(
            title,
            textAlign: TextAlign.center,
            style: text.headlineSmall?.copyWith(
              fontFamily: type.display,
              fontWeight: type.displayFontWeight,
              color: scheme.onSurface,
            ),
          ),
          if (amount != null) ...[
            const SizedBox(height: 12),
            Text(amount!, textAlign: TextAlign.center, style: money),
          ],
          if (subtitle != null) ...[
            const SizedBox(height: 8),
            Text(
              subtitle!,
              textAlign: TextAlign.center,
              style: text.bodyLarge?.copyWith(color: scheme.onSurfaceVariant),
            ),
          ],
        ],
        ),
      ),
    );
  }
}

/// A receipt card (web `<npt-receipt>`): a title, a list of label/value rows
/// (value right-aligned, amounts rendered in money style), a divider, and an
/// optional Share action. Each row is one spoken stop, "Fee: 1.500", never
/// a label and a value read as unrelated fragments. Theme-only, RTL-safe.
class NeptuneReceipt extends StatelessWidget {
  final String title;
  final List<({String label, String value})> rows;
  final VoidCallback? onShare;

  /// The Share button's label. Defaults to the host's word via
  /// [NeptuneA11yStrings.share].
  final String? shareLabel;

  const NeptuneReceipt({
    super.key,
    required this.title,
    required this.rows,
    this.onShare,
    this.shareLabel,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final shape = Theme.of(context).extension<NptShape>()!;
    final type = Theme.of(context).extension<NptType>()!;
    final text = Theme.of(context).textTheme;
    final money = NeptuneTheme.moneyStyle(context, base: text.bodyLarge)
        .copyWith(color: scheme.onSurface);

    return Container(
      decoration: BoxDecoration(
        color: scheme.surfaceContainerLowest,
        borderRadius: shape.rLg,
        border: Border.all(color: scheme.outlineVariant),
      ),
      padding: const EdgeInsetsDirectional.all(20),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: [
          Semantics(
            header: true,
            child: Text(
              title,
              style: text.titleLarge?.copyWith(
                fontFamily: type.display,
                fontWeight: type.displayFontWeight,
                color: scheme.onSurface,
              ),
            ),
          ),
          const SizedBox(height: 16),
          for (final row in rows)
            Padding(
              padding: const EdgeInsetsDirectional.only(bottom: 12),
              child: Semantics(
                label: '${row.label}: ${row.value}',
                excludeSemantics: true,
                child: Row(
                  crossAxisAlignment: CrossAxisAlignment.baseline,
                  textBaseline: TextBaseline.alphabetic,
                  children: [
                    Text(
                      row.label,
                      style: text.bodyMedium
                          ?.copyWith(color: scheme.onSurfaceVariant),
                    ),
                    const SizedBox(width: 16),
                    Expanded(
                      child: Text(
                        row.value,
                        textAlign: TextAlign.end,
                        style: money,
                      ),
                    ),
                  ],
                ),
              ),
            ),
          Divider(height: 1, thickness: 1, color: scheme.outlineVariant),
          if (onShare != null) ...[
            const SizedBox(height: 16),
            NeptuneButton(
              label: shareLabel ?? NeptuneAccessibility.of(context).share,
              icon: Icons.ios_share,
              variant: NeptuneButtonStyle.tonal,
              onPressed: onShare,
            ),
          ],
        ],
      ),
    );
  }
}

File: packages/neptune_kmp_ui/odyssey-compose-ui/src/commonMain/kotlin/ly/neptune/odyssey/ui/components/NeptuneReceipt.kt
// © 2026 Neptune.Fintech (neptune.ly) · Neptune Odyssey Community License v1.0
//
// The transfer outcome pair: the centred success hero and the itemised
// receipt card. Web counterparts: `<npt-success>` / `<npt-receipt>` ·
// Flutter: neptune_receipt.dart. The Dart success hero is a STATIC disc +
// glyph (it does not compose NeptuneStatusMotion) — flows that want the
// animated hand-off compose [NeptuneStatusMotion] themselves. Theme-only,
// RTL-safe.

package ly.neptune.odyssey.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Icon
import androidx.compose.material3.LocalContentColor
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.Immutable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import ly.neptune.odyssey.ui.glyphs.NptFintechGlyphs
import ly.neptune.odyssey.ui.glyphs.NptStatusGlyphs
import ly.neptune.odyssey.ui.theme.NeptuneTheme
import ly.neptune.odyssey.ui.theme.rememberNeptuneFontFamily

/**
 * A centred success hero: a large success-coloured 96dp disc with a 56dp
 * [icon] slot (check-circle glyph by default), a display-font title, an
 * optional big [amount] in money style, and an optional [subtitle].
 *
 * Web counterpart: `<npt-success>` · Flutter: `NeptuneSuccess`.
 */
@Composable
public fun NeptuneSuccess(
    title: String,
    modifier: Modifier = Modifier,
    amount: String? = null,
    subtitle: String? = null,
    icon: (@Composable () -> Unit)? = null,
) {
    val scheme = MaterialTheme.colorScheme
    val npt = NeptuneTheme.colors
    val typography = MaterialTheme.typography
    val money = NeptuneTheme.moneyStyle(base = typography.displaySmall)
        .copy(color = scheme.onSurface, fontWeight = FontWeight.W700)

    Column(
        modifier = modifier.padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            modifier = Modifier
                .size(96.dp)
                .clip(NeptuneTheme.shape.rFull)
                .background(npt.successContainer),
            contentAlignment = Alignment.Center,
        ) {
            CompositionLocalProvider(LocalContentColor provides npt.success) {
                Box(Modifier.size(56.dp), contentAlignment = Alignment.Center) {
                    if (icon != null) {
                        icon()
                    } else {
                        Icon(
                            NptStatusGlyphs.successCheck,
                            contentDescription = null,
                            modifier = Modifier.fillMaxSize(),
                        )
                    }
                }
            }
        }
        Spacer(Modifier.height(16.dp))
        // headlineSmall already rides the brand display face at the brand
        // display weight (the Dart copyWith is a no-op restated).
        Text(
            text = title,
            textAlign = TextAlign.Center,
            style = typography.headlineSmall.copy(color = scheme.onSurface),
        )
        if (amount != null) {
            Spacer(Modifier.height(12.dp))
            Text(
                text = NeptuneTheme.formatDigits(amount),
                textAlign = TextAlign.Center,
                style = money,
            )
        }
        if (subtitle != null) {
            Spacer(Modifier.height(8.dp))
            Text(
                text = subtitle,
                textAlign = TextAlign.Center,
                style = typography.bodyLarge.copy(color = scheme.onSurfaceVariant),
            )
        }
    }
}

/** One label/value line of a [NeptuneReceipt] (the Dart record row). */
@Immutable
public class NeptuneReceiptRow(
    public val label: String,
    public val value: String,
)

/**
 * A receipt card: a display-font title, a list of label/value rows (value
 * end-aligned in the money style), a hairline divider, and an optional Share
 * action (tonal [NeptuneButton]).
 *
 * Web counterpart: `<npt-receipt>` · Flutter: `NeptuneReceipt`.
 */
@Composable
public fun NeptuneReceipt(
    title: String,
    rows: List<NeptuneReceiptRow>,
    modifier: Modifier = Modifier,
    onShare: (() -> Unit)? = null,
) {
    val scheme = MaterialTheme.colorScheme
    val typography = MaterialTheme.typography
    val type = NeptuneTheme.type
    val shape = NeptuneTheme.shape.rLg
    val display = rememberNeptuneFontFamily(type.display)
    val money = NeptuneTheme.moneyStyle(base = typography.bodyLarge)
        .copy(color = scheme.onSurface)

    Column(
        modifier = modifier
            .clip(shape)
            .background(scheme.surfaceContainerLowest)
            .border(1.dp, scheme.outlineVariant, shape)
            .padding(20.dp),
    ) {
        Text(
            text = title,
            style = typography.titleLarge.copy(
                fontFamily = display,
                fontWeight = type.displayFontWeight,
                color = scheme.onSurface,
            ),
        )
        Spacer(Modifier.height(16.dp))
        for (row in rows) {
            Row(Modifier.padding(bottom = 12.dp)) {
                Text(
                    text = row.label,
                    style = typography.bodyMedium.copy(color = scheme.onSurfaceVariant),
                    modifier = Modifier.alignByBaseline(),
                )
                Spacer(Modifier.width(16.dp))
                Text(
                    text = NeptuneTheme.formatDigits(row.value),
                    textAlign = TextAlign.End,
                    style = money,
                    modifier = Modifier.weight(1f).alignByBaseline(),
                )
            }
        }
        // Hairline divider (the Dart 1dp outline-variant Divider).
        Box(Modifier.fillMaxWidth().height(1.dp).background(scheme.outlineVariant))
        if (onShare != null) {
            Spacer(Modifier.height(16.dp))
            NeptuneButton(
                label = "Share",
                onClick = onShare,
                variant = NeptuneButtonStyle.Tonal,
                icon = {
                    Icon(
                        NptFintechGlyphs.share,
                        contentDescription = null,
                        modifier = Modifier.size(20.dp),
                    )
                },
            )
        }
    }
}
